core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/different/DifferentOpExecConf.scala

```diff
@@ -0,0 +1,52 @@
+package edu.uci.ics.texera.workflow.operators.different
+
+import edu.uci.ics.amber.engine.architecture.breakpoint.globalbreakpoint.GlobalBreakpoint
+import edu.uci.ics.amber.engine.architecture.controller.Workflow
+import edu.uci.ics.amber.engine.architecture.deploysemantics.deploymentfilter.UseAll
+import edu.uci.ics.amber.engine.architecture.deploysemantics.deploystrategy.RoundRobinDeployment
+import edu.uci.ics.amber.engine.architecture.deploysemantics.layer.WorkerLayer
+import edu.uci.ics.amber.engine.common.Constants
+import edu.uci.ics.amber.engine.common.virtualidentity.{
+  ActorVirtualIdentity,
+  LayerIdentity,
+  LinkIdentity,
+  OperatorIdentity
+}
+import edu.uci.ics.amber.engine.operators.OpExecConfig
+
+//TODO: Mostly copied from HashJoinOpExecConf. Not quite understand some (all) of them.
+class DifferentOpExecConf[K](
```

Comment:
I think the problem with Difference is that, it is not symmetric. For intersect, it is symmetric, because you do not need to know whether a tuple comes from left or right. You just need to collect all the tuples that exist in both sides, without specifying the exact side.  For distinct, you just put everything in a set and return an iterator to perform deduplication, so you still do not need the exact left-or-right.
But it is necessary for DifferentOpExecConf to know which is left and which ls right, because you are actually specifying that, you want all tuples that are on the left side, but are not on the right side, which is similar to HashJoin. HashJoin knows which input table is build and which is probe by using HashJoinOpExecConf, ```edu.uci.ics.texera.workflow.operators.hashJoin.HashJoinOpExecConfig#buildTable```. However, ```HashOpExecConf``` does not provide such attribute. You can still extend ```HashOpExecConf```, but I think they are more or less the same. I actually wonder if there is a way to work around the step to identify left and right, but I did not figure it out. So I kept ```DifferentOpExecConf``` to use ```edu.uci.ics.amber.engine.operators.OpExecConfig#inputToOrdinalMapping``` in ```edu.uci.ics.texera.workflow.operators.difference.DifferenceOpExecConf#checkStartDependencies``` to specify whith LinkIdentity is the right side.

Reply:
Understood, we'll keep it this way for now.